SharedProjects/BabylonExport.Entities/BabylonVertexData.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Runtime.Serialization;
+
+namespace BabylonExport.Entities
+{
+
+    [DataContract]
+    public class BabylonVertexData : IBabylonMeshData
```

Comment:
If the goal here is to abstract the geometry information from the rest of the mesh information, should we remove the geometry-related fields from BabylonMesh, and instead use a facade to the BabylonGeometry when serializing?

Reply:
Agree for an SDK, but here i stick with the original design idea which is to have Entities dedicated to serialization. Babylon format, introduce the geometries fields late into version 2 (i guess remember), and for compatibility, the file format is keeping the both. David Catuhe or other Guru  will be able to tell us if this is still in use into Babylon. My concern was, related to our first conversation, was to keep the change "minimal". I will be happy to explore this if you think it's mandatory.

Reply:
I'm not sure if I'm being clear here.

Imagine we have two meshes, one original and a clone of that original mesh in scene:

when serialized even though we are using the IBabylonMeshData, interface, but we are also not adding any additional logic to the glTF serializer to handle serialization of meshes with GeometryData only, so I'm concerned that the second mesh may not be properly serialized, as we will be using those nulled fields.

Personally I don't believe that we should limit our design for backwards compatibility here. @deltakosh, your input would be appreciated as well.

If we decide to go all in with GeometryData, we should go all in and use this as our storage of geometry, else we're complicating our design with additional usecases which may or may not be valid.
@Drigax

Reply:
I put all (as far i know ) the mandatory logic in place inside the GLTF part of the exporter.  I asked a lot of question inside the issue but no response :-), then i move forward and also provide sample and result at the end of the issue.
I'm agree that this enhancement is quite intrusive and this is why i put it as "experimental". 
So, Let's continue this interesting discussion :-).

Reply:
Cool, missed ALLL of those changes, thanks. Since you're handling both cases relatively elegantly in the glTF exporter, I'm much less worried. Lets check this in and revisit if we run into any issues in the future.